engines/producers/app/policies/producers/product_policy.rb

```diff
@@ -0,0 +1,61 @@
+module Producers
+  class ProductPolicy < Shared::ApplicationPolicy
+    def create?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    def show?
+      user_membership_for?(user) || group_membership_for?(user)
+    end
+
+    def update?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    def destroy?
+      user_admin_membership_for?(user) || group_admin_membership_for?(user)
+    end
+
+    private
+
+    def user_membership_for?(user)
```

Comment:
We could even move it to the model in the specific engine, like `producer.is_admin?(user)`